draft-ietf-cose-x509.xml

```diff
@@ -121,7 +121,7 @@ obtained from any of these methods still need to validate it.
       -->
           </t>
           <t>
-				Unless it is known to both sender and recipient that proof-of-possession of the subject's private key was required for certificate issuance, the end-entity certificate MUST be integrity protected by COSE. This can e.g. be done by sending the header parameter in the protected header, sending a x5bag in the unprotected header combined with a x5t in the protected header, or including the end-entity certificate in the external_aad as is done in EDHOC. 
+				Unless it is known to both sender and recipient that proof-of-possession of the subject's private key was required for certificate issuance, the end-entity certificate MUST be integrity protected by COSE. This can e.g. be done by sending the header parameter in the protected header, sending a x5bag in the unprotected header combined with a x5t in the protected header, or including the end-entity certificate in the external_aad. 
```

Comment:
"the external_aad" itself is something of an EDHOC reference; phrasing as "including the end-entity certificate in an integrity protection mechanism at the application layer" might be more generic.
(applies throughout but I will just say it once)

Reply:
> "the external_aad" itself is something of an EDHOC reference;

Not really, RFC 8152 has the a whole section 4.3 describing Externally Supplied Data, i.e. external_aad. external_aad is very much a COSE construct. I think qhat you suggest would be yet another way that would be outside of COSE.

Reply:
We could change the text to:

"This can e.g. be done by sending the header parameter in the protected header, sending a x5bag in the unprotected header combined with a x5t in the protected header, including the end-entity certificate in the external_aad, or ncluding the end-entity certificate in an integrity protection mechanism at the application layer"

But anything that is not using the Externally Supplied Data / external_aad mechanism provided by COSE would not longer meet the requirement "The end-entity certificate MUST be integrity protected by COSE.". That requirement would then have to change.

Reply:
I would suggest keeping the text as is. Or do you want to open it up for integrity protection outside of COSE. That probably comes with additional problems on how to bind things to the COSE message.